### ISSUE TITLE
Text formatting in the body of incoming email

### ISSUE THREAD

**Issue by style4** [salesagility/SuiteCRM#7801]
#### Issue
I'm new to SuiteCRM and having some trouble with text formatting in the body of incoming email. This is both for personal and group mail accounts. If there is any formatting or any attachments in the incoming mail, then "junk characters" showing up in the body.

#### Expected Behavior
More readable text

#### Steps to Reproduce
1. Clean install
2. Create one personal inbound email account
3. Send an email from Mail on Mac OS X with one word in bold to this email account
4. Look at the body from within SuiteCRM (see below)

![test1](https://user-images.githubusercontent.com/3805447/64163245-2e825400-ce41-11e9-9fc8-562cba50aff6.jpg)

When sending an email without any formatting everything is OK  (see below)

![test2](https://user-images.githubusercontent.com/3805447/64163313-4eb21300-ce41-11e9-9c70-4f08fcc140fe.jpg)

#### Your Environment
* SuiteCRM Version used: Tested on both SuiteCRM 7.10.20 LTS and SuiteCRM 7.11.8 (both clean installations, one with Swedish Language pack)
* Browser name and version: Chrome Version 76.0.3809.132 (64-bit)
* Server: Tested both locally on my Mac (PHP 7.1) and on external host with Linux (PHP 7.2). Apache on both machines.
* Charset in SuiteCRM settings: I've tested both UTF-8 and ISO 8859-1 (Latin 1)
Database: MySQL 5.7.21 (UTF-8)

**Comment by labcerouno** [comment]
We are having the same issue since upgrading to version 7.11.8 (from v7.11.4). I think this is High Priority because almost 100% of emails we receive are HTML. We can't read almost any e-mail since then. I can help debugging this issue, if someone indicates me the steps to do so. We need a fix, please!

Thank you very much. Best regards

**Comment by pgorod** [comment]
Isn't this issue the same as #7880 ?

If it is, it's being worked on so I expect a fix is coming soon....

@Dillon-Brown 👀

**Comment by Dillon-Brown** [comment]
@labcerouno @pgorod Yes it's the same issue as #7880. I've put up a fix here which should resolve this: #7922 but it goes a bit further in starting to refactor how we handle inbound emails with the goal of avoiding these HTML formatting and encoding issues.

**Comment by labcerouno** [comment]
Thank you @Dillon-Brown .
It seems to be working fine. I'll test a bit more during the next days. 

To anyone who's trying to apply the fix, keep in mind that if you have the official package, you need to add some dependencies on `composer.json` (`mail-mime-parser` and `ext-imap`). Do not replace your `composer.json` file by dillon's. Only the PHPs on this fix.

**Comment by Mausino** [comment]
![2019-09-12_0138_1](https://user-images.githubusercontent.com/5773777/65587122-9ed64e00-df85-11e9-8662-e4be34c3536d.png)

We've reverted InboundEmail.php file to version before upgrade so now should work for new emails. We can't repair emails imported in version 7.11.8, however seems that not all emails have this problem.

**Comment by johnM2401** [comment]
Hey!

Thank you all for raising this and for your discussions on this.

As noted above,  This has been resolved via the PR: https://github.com/salesagility/SuiteCRM/pull/7922.

I've given a few tests in the latest release and have not been able to replicate this issue.

<br>

I'm going to close this off for now.
however, if you're still seeing this issue in the latest releases, please don't hesitate to get in touch.

Thank you again!
